File: Project/MyProject/MyProject/Content/MyAsset/Lua/MyLua/Libs/DataStruct/NoOrPriorityList/PriorityList/PriorityProcessObject.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

-- 优先级队列对象
local M = GlobalNS.Class(GlobalNS.GObject);
M.clsName = "PriorityProcessObject";
GlobalNS[M.clsName] = M;

function M:ctor()
	self.mPriorityObject = nil;
	self.mPriority = 0.0;
end

return M;

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/DataStruct/NoOrPriorityList/PriorityList/PriorityProcessObject.cs
namespace SDK.Lib
{
    /**
     * @brief 优先级队列对象
     */
    public class PriorityProcessObject : GObject
    {
        public INoOrPriorityObject mPriorityObject;
        public float mPriority;

        public PriorityProcessObject()
        {
            
        }

        override protected void onPutInPool()
        {
            this.mPriorityObject = null;
            this.mPriority = 0.0f;
        }
    }
}